cmd/snap-update-ns/utils.go

```diff
@@ -346,108 +318,103 @@ func splitIntoSegments(name string) ([]string, error) {
 // The uid and gid are used for the fchown(2) system call which is performed
 // after each segment is created and opened. The special value -1 may be used
 // to request that ownership is not changed.
-func (sec *Secure) MkdirAll(name string, perm os.FileMode, uid sys.UserID, gid sys.GroupID) error {
-	logger.Debugf("secure-mkdir-all %q %v %d %d", name, perm, uid, gid)
-
+func (sec *Secure) MkdirAll(path string, perm os.FileMode, uid sys.UserID, gid sys.GroupID) error {
+	if path != filepath.Clean(path) {
+		// TODO: Remove this, it is only to make tests happy
+		return fmt.Errorf("cannot split unclean path %q", path)
+	}
```

Comment:
Why wouldn't this also be an error? I mean, I realize you are cleaning below, but is there a case you would want to proceed with an unclean path?

Reply:
This is more about "splitting" than the clean path vs unclean path. I will remove the TODOs and let subsequent patches speak for themselves. The check is kept but the error message is saner.